coruscant/src/containers/UserSettings.js

```diff
@@ -0,0 +1,69 @@
+import React, { Component } from 'react'
+import { Link, Redirect, withRouter } from 'react-router-dom'
+import { Tab, Tabs } from 'material-ui'
+import PropTypes from 'prop-types'
+
+import AccountSettings from '../components/user-settings-tabs/AccountSettings'
+import InvitesSettings from '../components/user-settings-tabs/InvitesSettings'
+import NotificationSettings from '../components/user-settings-tabs/NoficiationSettings'
+import PageWrap from '../components/PageWrap'
+import ProfileSettings from '../components/user-settings-tabs/ProfileSettings'
+
+
+const tabValues = ['profile', 'invites', 'accounts', 'notification']
+
+class UserSettings extends Component {
+  constructor(props) {
+    super(props)
+    this.state = {
+      value: props.match.params.tabValue,
+    }
+  }
+
+  handleChange = (value) => {
+    this.setState({
+      value
+    })
+  }
+
+  render() {
+    // Catch any bad paths in the settings subdirectory
+    const currentTabValue = this.props.match.params.tabValue
```

Comment:
Move this to componentDidMount. Also avoid using redirect components here. There is a history prop that is passed to all route components(The component loaded for a specific route that you specify in the index.js) that has functions - push, replace, etc to manipulate routes